src/libs/ReportUtils.ts

```diff
@@ -3465,13 +3471,54 @@ function shouldHideReport(report: OnyxEntry<Report>, currentReportId: string): b
 }
 
 /**
- * Takes several pieces of data from Onyx and evaluates if a report should be shown in the option list (either when searching
- * for reports or the reports shown in the LHN).
+ * Checks to see if a report's parentAction is a money request that contains a violation
+ */
+function doesTransactionThreadHaveViolations(report: Report, transactionViolations: TransactionViolations, reportActions: OnyxCollection<ReportActions>): boolean {
+    const resolvedParentReportAction = reportActions?.[`${report.parentReportID}`]?.[`${report.parentReportActionID}`];
+    if (!resolvedParentReportAction) {
+        return false;
+    }
+    if (resolvedParentReportAction.actionName !== CONST.REPORT.ACTIONS.TYPE.IOU) {
+        return false;
+    }
+    const {IOUTransactionID, IOUReportID} = resolvedParentReportAction?.originalMessage;
```

Comment:
Is this syntax right `reportActions?.[..`? Regardless, I think we can remove the `?` sine reportActions is no longer optional, and `resolvedParentReportAction` is a leftover from when we were passing `parentReportAction` as a param, so let's update it
```suggestion
    const parentReportAction = reportActions[`${report.parentReportID}`]?.[`${report.parentReportActionID}`];
    if (!parentReportAction) {
        return false;
    }
    if (parentReportAction.actionName !== CONST.REPORT.ACTIONS.TYPE.IOU) {
        return false;
    }
    const {IOUTransactionID, IOUReportID} = parentReportAction?.originalMessage;
```

Reply:
reportActions is of type `OnyxCollection<ReportActions>` which can be null.

![CleanShot 2023-12-19 at 13 46 37@2x](https://github.com/Expensify/App/assets/22041394/6c2786ff-43f5-4418-8df9-7ee85afe27c7)